### PR TITLE
Update for Contributors library guidelines

### PR DESCRIPTION
**Pull request by thomashoneyman**
This pull request is part of an effort to update and standardize the Contributors libraries according to the [Library Guidelines](https://github.com/purescript-contrib/governance/blob/main/library-guidelines.md). Specifically, it:

1. Adjusts the files and repository structure according to the [repository structure](https://github.com/purescript-contrib/governance/blob/main/library-guidelines.md#repository-structure) section of the guidelines, which includes standard pr templates, issue templates, CI in GitHub Actions, ensures the project uses Spago, and so on.
2. Updates the README and documentation according to the [documentation](https://github.com/purescript-contrib/governance/blob/main/library-guidelines.md#documentation) section of the guidelines. This is a first step towards ensuring Contributors libraries have adequate module documentation, READMEs, a docs directory, and tests (even if just usage examples) in a `test` directory.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -1,28 +1,29 @@
...
-    "no-return-assign": 2,
-    "no-unused-expressions": 2,
-    "no-use-before-define": 2,
-    "radix": [2, "always"],
-    "indent": [2, 2],
-    "quotes": [2, "double"],
-    "semi": [2, "always"]
+    "block-scoped-var": "error",
+    "consistent-return": "error",
+    "eqeqeq": "error",
```
Comment (JordanMartinez) [inline .eslintrc.json:8]: The original file has `[2, "smart"]`. Do we need to keep that?

File: .eslintrc.json [hunk 2]
```diff
@@ -1,28 +1,29 @@
...
+    "no-new": "error",
+    "no-param-reassign": "error",
+    "no-return-assign": "error",
+    "no-sequences": "error",
+    "no-unused-expressions": "error",
+    "no-use-before-define": "error",
+    "no-undef": "error",
+    "no-eq-null": "error",
+    "radix": ["error", "always"],
+    "indent": ["error", 2, { "SwitchCase": 1 }],
```
Comment (JordanMartinez) [inline .eslintrc.json:24]: Original file had `[2, 2]`. Why was this change made?

**Review by JordanMartinez — APPROVED** [review]

File: .eslintrc.json (hunk 1 above)
Comment (thomashoneyman) [reply .eslintrc.json:8]: It's been overwritten to use the standard, which is a bit stricter, but makes it more explicit when you're choosing to compare against null / undefined.

File: .eslintrc.json (hunk 2 above)
Comment (thomashoneyman) [reply .eslintrc.json:24]: This rule was based on the same rule used in `Aff` and it's just changed here for consistency.

Comment (JordanMartinez) [reply .eslintrc.json:24]: Gotcha. Then let's merge this in.

**Pull request merged**